src/components/scatterplot/ScatterplotSubscriber.js

```diff
@@ -9,86 +11,99 @@ import {
+  const {
+    onReady,
+    mapping,
+    uuid = null,
+    children,
+    view,
+    removeGridComponent,
+    theme,
+  } = props;
+
+  const [cells, setCells] = useState({});
+  const [selectedCellIds, setSelectedCellIds] = useState(new Set());
+  const [cellColors, setCellColors] = useState(null);
+  const [cellRadiusScale, setCellRadiusScale] = useState(0.2);
 
-  // eslint-disable-next-line camelcase
-  UNSAFE_componentWillMount() {
-    this.cellsAddToken = PubSub.subscribe(
-      CELLS_ADD, this.cellsAddSubscriber.bind(this),
+  const onReadyCallback = useCallback(onReady, []);
+
+  useEffect(() => {
+    const cellsAddToken = PubSub.subscribe(
+      CELLS_ADD, (msg, data) => {
+        setCells(data);
+      },
     );
-    this.cellsColorToken = PubSub.subscribe(
-      CELLS_COLOR, this.cellsColorSubscriber.bind(this),
+    const cellsColorToken = PubSub.subscribe(
+      CELLS_COLOR, (msg, data) => {
+        setCellColors(data);
+      },
     );
-    this.cellsSelectionToken = PubSub.subscribe(
-      CELLS_SELECTION, this.cellsSelectionSubscriber.bind(this),
+    const cellsSelectionToken = PubSub.subscribe(
+      CELLS_SELECTION, (msg, data) => {
+        setSelectedCellIds(data);
+      },
     );
-    this.cellSetsViewToken = PubSub.subscribe(
-      CELL_SETS_VIEW, this.cellsSelectionSubscriber.bind(this),
+    const cellSetsViewToken = PubSub.subscribe(
+      CELL_SETS_VIEW, (msg, data) => {
+        setSelectedCellIds(data);
+      },
     );
-  }
-
-  componentDidMount() {
-    const { onReady } = this.props;
-    onReady();
-  }
-
-  componentWillUnmount() {
-    PubSub.unsubscribe(this.cellsAddToken);
-    PubSub.unsubscribe(this.cellsColorToken);
-    PubSub.unsubscribe(this.cellsSelectionToken);
-    PubSub.unsubscribe(this.cellSetsViewToken);
-  }
-
-  cellsSelectionSubscriber(msg, cellIds) {
-    this.setState({ selectedCellIds: cellIds });
-  }
+    onReadyCallback();
+    return () => {
+      PubSub.unsubscribe(cellsAddToken);
+      PubSub.unsubscribe(cellsColorToken);
+      PubSub.unsubscribe(cellsSelectionToken);
+      PubSub.unsubscribe(cellSetsViewToken);
+    };
+  }, [onReadyCallback, mapping]);
 
-  cellsColorSubscriber(msg, cellColors) {
-    this.setState({ cellColors });
-  }
+  // After cells have loaded or changed,
+  // compute the cell radius scale based on the
+  // extents of the cell coordinates on the x/y axes.
+  useEffect(() => {
+    if (cells) {
+      const cellCoordinates = Object.values(cells)
+        .map(c => c.mappings[mapping]);
+      const xExtent = extent(cellCoordinates, c => c[0]);
+      const yExtent = extent(cellCoordinates, c => c[1]);
+      const xDifference = xExtent[1] - xExtent[0];
+      const yDifference = yExtent[1] - yExtent[0];
+      const area = xDifference * yDifference;
 
-  cellsAddSubscriber(msg, cells) {
-    this.setState({ cells });
-  }
+      const newScale = clamp(area / 7000, 0, 0.2);
+      if (newScale && !Number.isNaN(newScale)) {
```

Comment:
I think `NaN` is falsely, so can this be simplified? 
```suggestion
      if (newScale) {
```